src/Symfony/Component/HttpFoundation/UrlParser/Url.php

```diff
@@ -0,0 +1,80 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\HttpFoundation\UrlParser;
+
+/**
+ * @author Alexandre Daubois <alex.daubois@gmail.com>
+ */
+final class Url implements \Stringable
+{
+    public function __construct(
+        public string $scheme,
+        public ?string $user = null,
+        public ?string $pass = null,
```

Comment:
Could we use `password` here ?

Reply:
We could, but this would make impossible to do the following:

```php
$url = new Url(...parse_url($url));
```

This is why I kept `pass` here

Reply:
the goal of the new code should be to completely replace using parse_url, so this argument is void sorry. (parse_url is broken)

Reply:
Oh yes, of course 🤦 Updated accordingly